### ISSUE TITLE
Locked because of GeoIP Policy

### ISSUE THREAD

**Issue by bnounours** [goauthentik/authentik#12214]
I added a GeoIP policy to the default authentication flow. I choose France as country.
Now I cannot login anymore as I get the error "GeoIP: client IP address not found in City database." when I connect locally. I tried from my mobile phone and get the same result :(

I tried https://docs.goauthentik.io/docs/troubleshooting/login to unblock but finally get the same result :(

```
{
    "action": "policy_exception",
    "auth_via": "unauthenticated",
    "client_ip": "192.168.X.X",
    "context": {
        "binding": {
            "app": "authentik_policies",
            "model_name": "policybinding",
            "name": "Binding from Flow Welcome to authentik! (default-authentication-flow) #0 to Policy V\u00e9rification de la provenance de l'IP",
            "pk": "773e7980d24b4719a976fafb5bf667ab"
        },
        "http_request": {
            "args": {
                "next": "/"
            },
            "method": "GET",
            "path": "/api/v3/flows/executor/default-authentication-flow/",
            "request_id": "d7c6127820c445b4ae862fa8672e3f99",
            "user_agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/133.0"
        },
        "message": "Traceback (most recent call last):\nauthentik.policies.geoip.exceptions.GeoIPNotFoundException: GeoIP: client IP address not found in City database.",
        "policy_uuid": "b3d0322f75664297b97aaa9702f7bc76",
        "request": {
            "context": {
                "asn": null,
                "geoip": null
            },
            "debug": false,
            "http_request": null,
            "obj": {
                "app": "authentik_flows",
                "model_name": "flow",
                "name": "Welcome to authentik!",
                "pk": "0e1c3278711843849c33c5eebe208430"
            },
            "user": {
                "email": "",
                "is_anonymous": true,
                "pk": 2,
                "username": "AnonymousUser"
            }
        }
    },
    "domain_url": "yyy.ttttt.com",
    "event": "Created Event",
    "host": "yyy.ttttt.com",
    "level": "info",
    "logger": "authentik.events.models",
    "pid": 151,
    "request_id": "d7c6127820c445b4ae862fa8672e3f99",
    "schema_name": "public",
    "timestamp": "2024-11-27T23:22:09.209279",
    "user": {
        "email": "",
        "is_anonymous": true,
        "pk": 2,
        "username": "AnonymousUser"
    }
}
```

I also tried to deactivate geoip, but doesn't seems to work : https://docs.goauthentik.io/docs/sys-mgmt/ops/geoip#deactivating-geoip

How can I remove / disable the policy or reset the flow to be able to log again instead of restoring the database ?

Can you help me ?

**Comment by BeryJu** [comment]
Using https://docs.goauthentik.io/docs/troubleshooting/login skips any flows and policies so that will always work, what error did you get?

The issue seems to come from the fact that authentik doesn't know the real client IP, as in the log you can see a 192.168 IP being shown.

**Comment by solidDoWant** [comment]
I recently ran into the above as well.

Could an option be added to geoip policies to ignore certain CIDRs (such as local address ranges)? As it currently stands, a geoip policy bound to a flow will always cause the flow to fail when attempting to log in from private IP space (and probably others, such as link local addresses or bogon IP space).

**Comment by bnounours** [comment]
> Using https://docs.goauthentik.io/docs/troubleshooting/login skips any flows and policies so that will always work, what error did you get?
> 
> The issue seems to come from the fact that authentik doesn't know the real client IP, as in the log you can see a 192.168 IP being shown.

Hello @BeryJu, indeed I am connecting from private network so the IP is the good one.

I tried this (https://docs.goauthentik.io/docs/troubleshooting/login ) but it applies policies so I was still blocked.

I finally manage to connect and remove the policy from external connection. I would like to add this policy but I am afraid of blocking myself again as the private IP is checked by GeoIP policy too

**Comment by vidonnus** [comment]
This is possible using an "Expression Policy" like so,
```
if ip_address(ak_client_ip) in ip_network('192.168.0.0/24'):
  return True
else:
  return context["geoip"]["country"] == "US"
```
where the country code is the two character [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) alpha code for the country.

Documentation on other potential variables can be found here: https://docs.goauthentik.io/docs/customize/policies/expression#comparing-ip-addresses

Issue can be closed.

**Comment by dominic-r** [comment]
We're closing this in favor of this parent issue: #25279
